export_contact_lists.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+import csv
+import json
+
+from core_data_modules.cleaners import Codes
+from core_data_modules.logging import Logger
+from core_data_modules.traced_data.io import TracedDataJsonIO
+from id_infrastructure.firestore_uuid_table import FirestoreUuidTable
+from storage.google_cloud import google_cloud_utils
+
+from src.lib import PipelineConfiguration
+from src.lib.code_schemes import CodeSchemes
+
+Logger.set_project_name("IOM")
+log = Logger(__name__)
+
+IOM_LOCATIONS = {"cabudwaaq", "gaalkacyo", "dhuusamarreeb"}
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Generates lists of phone numbers of previous CSAP respondents who  "
+                                                 "were labelled as living in one of the IOM target districts")
+
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file")
+    parser.add_argument("traced_data_paths", metavar="traced-data-paths", nargs="+",
+                        help="Paths to the traced data files (either messages or individuals) to extract phone "
+                             "numbers from")
+    parser.add_argument("csv_output_file_path", metavar="csv-output-file-path",
+                        help="Path to a CSV file to write the contacts from the districts of interest to. "
+                             "Exported file is in a format suitable for direct upload to Rapid Pro")
+
+    args = parser.parse_args()
+
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    traced_data_paths = args.traced_data_paths
```

Comment:
@as2388  QQ - How do pass this? one project traced data at a time?

Reply:
You'd do something like `python export_contact_lists.py credentials.json config.json traced-data-1.jsonl traced-data-2.jsonl traced-data-3.jsonl output.csv`
(which I need because CSAP requires pulling data from a variable number of previous projects).

I think your question comes from not having seen the nargs="+" setting of argparse before. If there's something else I should do to make this easier to understand though, let me know

Reply:
Yes `nargs="+"` was new to me but it is now clear. Thankyou.